File: proj0/NBody.java
import java.awt.font.NumericShaper;
/** this class will run the Planet class */
public class NBody {

    /** home directory for images of planets and background stars*/
    public static String dirFile = "./images/";
    public static String starFieldBackground = "starfield.jpg";

    /** used to read the radius in the ./data/planet.txt file. Reads the lines */
    public static double readRadius(String fname ){
        In pathFile = new In(fname);
        int numPlanets = pathFile.readInt();
        double radius = pathFile.readDouble();
        return radius;
    }
    /**  reads lines in .txt file that contain planet info */
   public static Planet[] readPlanets(String fname){
       In pathFile = new In(fname);
       int numPlanets = pathFile.readInt();
       double radius = pathFile.readDouble();
       int dummyIndx = 0;
       Planet[] planets = new Planet[numPlanets];
       while (dummyIndx < numPlanets ){
           planets[dummyIndx] = new Planet(0.0, 0.0, 0.0, 0.0, 0.0, "");
           planets[dummyIndx].xxPos = pathFile.readDouble();
           planets[dummyIndx].yyPos = pathFile.readDouble();
           planets[dummyIndx].xxVel = pathFile.readDouble();
           planets[dummyIndx].yyVel = pathFile.readDouble();
           planets[dummyIndx].mass = pathFile.readDouble();
           planets[dummyIndx].imgFileName = pathFile.readString();
           dummyIndx +=1;
       }
       return planets;

   }

   /** used to make animation. of planets. */
   public static void main(String[] args){

       double T = Double.parseDouble(args[0]);
       double dt = Double.parseDouble(args[1]);
       String filename = args[2];

       Planet[] planets = readPlanets(filename);
       for (Planet p: planets) {
           p.imgFileName = dirFile + p.imgFileName;
       }

       double radius  = readRadius(filename);
       //StdDraw.setCanvasSize(1000,800);
       StdDraw.setScale(-radius,radius);

       double time = 0;
       double[] xForces = new double[planets.length];
       double[] yForces = new double[planets.length];

       while (time < T){
           planetLoop(planets);
           for (int indx  = 0;indx  < planets.length;indx++){
               xForces[indx] = planets[indx].calcNetForceExertedByX(planets);
               yForces[indx] = planets[indx].calcNetForceExertedByY(planets);
           }
           int indx = 0;
           for(Planet p : planets){
               p.update(dt, xForces[indx], yForces[indx]);
               indx +=1;
           }
           time += dt;
       }

   }

   /** used to loop over and draw each planet in planet array */
   public static void planetLoop(Planet[] planets){
        StdDraw.clear();
        StdDraw.picture(0, 0, dirFile+starFieldBackground);
        //StdDraw.picture(0, 0, dirFile+starFieldBackground,2.50e+11*2,2.50e+11*2);
        for (Planet p: planets){
            //p.imgFileName = dirFile + p.imgFileName;
            p.draw();
        }
        StdDraw.show(10);
    }
}

File: PythonOOP/Proj0PY/NBody.py

import numpy as np
from Planet import Planet 
import sys
#from PyQt5 import  QtGui, QtWidgets, QtCore, QtHelp
import pygame
from pygame.locals import *

import time



class NBody(object):
	""" docstring for NBody
	* Datapath = "C:/Users/varga_000/Documents/cs61b/skeleton-sp17/proj0/data/"
	* starfield = "starfield.jpg"
	* AudioPath = "C:/Users/varga_000/Documents/cs61b/skeleton-sp17/proj0/audio/"
	* ExamplePaht = "C:/Users/varga_000/Documents/cs61b/skeleton-sp17/proj0/examples/"
	* imagesPath = "C:/Users/varga_000/Documents/cs61b/skeleton-sp17/proj0/images/"
	"""
	

	def readRadius(fname):
		data = open( fname)
		numPlanets = int(data.readline())
		radius = float(data.readline())
		return radius

	def readPlanets(fname):
		data = open( fname)
		numPlanets = int(data.readline())
		radius = float(data.readline())
		planetsList = [Planet(*data.readline().split()) for planet in range(numPlanets)]
		return planetsList

	def XScale2pix(width, x, Xmin, Xmax):
		return  width * ((x - Xmin)/ (Xmax - Xmin))

	def YScale2pix(height, y, Ymin, Ymax):
		return  height * ((Ymax - y)/ (Ymax - Ymin))

	def __main__():
		try:
			""" 
				* Command line requires Time(ms), dt(ms) and fname
				* EX)>>>python NBody.py Time dt data/planets.txt' 
			"""
			T, dt, fname = sys.argv[1:]
			#print(T, dt)
			T, dt = float(T), float(dt)
			imagesPath = "./images/"

			width, height = 800, 800

			
			def Window():
				radius = int(NBody.readRadius('./data/' + fname))
			
				#print(radius)
				planets = NBody.readPlanets('./data/' + fname)
				screen = pygame.display.set_mode((width, height))
				bkgImg = pygame.image.load(imagesPath + 'starfield.jpg')
				bkgImg = pygame.transform.scale(bkgImg,(width, height))

				
				time  = 0
				while time < T:
					xForces = []
					yForces = []
					screen.blit(bkgImg,(0,0))
					for plnt in planets:
						pi = pygame.image.load(imagesPath + plnt.imgFileName)
					
						x,y = NBody.XScale2pix(width, plnt.xxPos, -radius, radius) , NBody.YScale2pix( height, plnt.yyPos, -radius, radius)						
						
						screen.blit(pi,(x,y))

					for plnt in planets:
						xForces += [plnt.calcNetForceExertedByX(planets)]
						yForces += [plnt.calcNetForceExertedByY(planets)]
					for i in range(len(planets)):
						#print("{:.4E}".format(planets[i].xxPos), "{:.4E}".format(planets[i].yyPos), "{:.4E}".format(planets[i].xxVel), "{:.4E}".format(planets[i].yyVel), "{:.4E}".format(planets[i].mass), planets[i].imgFileName)
						planets[i].update(dt, xForces[i], yForces[i])
					#print()


					for event in pygame.event.get():
						if event.type == pygame.QUIT:
							sys.exit()
					time += dt
					pygame.display.update()
				#print(len(planets))
				#print(radius)
				for i in range(len(planets)):
					print("{:.4E}".format(planets[i].xxPos), "{:.4E}".format(planets[i].yyPos), "{:.4E}".format(planets[i].xxVel), "{:.4E}".format(planets[i].yyVel), "{:.4E}".format(planets[i].mass), planets[i].imgFileName)
				sys.exit()


			Window()








		except ValueError as err:
			print('ValueError: system args missing ','(ie. T , dt, fileName)')
			print('if running from command line use the following syntax: ')
			print('>>>python NBody.py 157788000.0 25000.0 data/planets.txt')
			pass





NBody.__main__()

